### DISCUSSION TITLE
in-game codex

### DISCUSSION THREAD

**Discussion by nsklaus** [cataclysmbn/Cataclysm-BN#3251] [General]
### Is your feature request related to a problem? Please describe.

C:BN like it's ancestor CDDA lacks a full loop in-game documentation for the players to use regarding the wide variety of objects, locations, NPCs, events and monsters that can be found in the game.
for example, the crafting system often times request availability of some materials, but give no clues about them.
it lacks a way to inspect individual components that are listed as needed/requested so we get to know what we are looking for, where to go to get them, or how to obtain them (e.g: by disassembling something else).
so we're tasked to find stuff we have no clues about. thinking about this kind of problem i thought it would be very handy to have an in-game codex, to centralize information about all the things we get to interact with.
get to know the objects and components better, as well as the various locations that can be visited, the NPCs and monster we encounter as well as the various events and also the data we gather in game. having such a system would prevent the player to have to pause the game and go look online using a browser to find out the infos they need. it could all be a part of the game and a fun one at that, since it would allow to build and expand the lore better. i describe below that idea in more details. 

### Describe the solution you'd like

having an in-game codex that could serve multiple purposes.

it could act as an always up to date and autogenerated source of documentation available directly in game. (and also exportable eventualy, in pdf or html, so no more outdated wiki. game and wiki doc would be in sync)
it could also serve as a good place to centralize and access various informations on the game lore and let it expand with dedicated UI panel. i will describe this idea a bit more below. also i made a few mockup of such a 'codex' panel if it existed, that is one possibility (idea) of how it could look. they are probably full of inaccuracies, don't mind that, they are just here for the general idea.

so, here are the mockups:

monsters:
![monsters](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/6810861/71352a46-a252-45a6-9994-31b26e4852fa)

survivors: 
![npcs](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/6810861/d9022ae4-bf76-491d-a0c3-e1328e80c1ee)

documents:
![documents](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/6810861/e2321aab-e248-4d7b-bf3c-4c448ac5a69c)

locations:
![locations](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/6810861/342252f3-fab6-42e0-ad2b-2b49212b17a9)

events:
![events](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/6810861/e63c8e06-a509-479d-93ba-8964d8e272f0)

and now i explain a bit some of the details:

The codex should use one image from currently selected tileset per item listed.
The image space could also be used to put ascii art text instead of picture ( for ncurse or ascii tiles users, and ascii artists )
Each monsters would get added to the codex, as they are encountered.
as an idea: maybe few encounter needed to get full monster details in the codex. (like studying monsters). each encounter yield additional data written in the codex entry for this monster, after a few encounter the page entry is fully shown.
Each npc gets added to the codex when they are encountered. a good place to remember who's who.. where you've met them in case you forget, what do they want, what do they need, their faction etc ..
Each document, when it is read, is filled in the codex. all newpapers, flyers, books..
Objects. all game objects should have their entry. so when you find something, and you don't know what it is you could also look at the codex as a documentation.
Locations too, each time a new location is visited, it gets added to the codex.
Events. keep track of overall progress and game events. a good place to find clear resume of the situation so far, instead of having it burried in the log mixed with everything else. like an overall log but only for importants events that matter saving. for example: fullfilled quests, achievements, parts of scenario, reached milestones, things the player noticed about this world. it would also be giving game's lore more visibility. seeing all those events would give at a glance, a sense of what this world became after the fall. thus increasing immersion and atmosphere.
There can be no 'doubles', and there should be only unique entries. Generic objects count for one (don't register 1million 'houses' or 'plastic bowl' or 'zombies' ... entries in the codex, unless they have something additional or particular about them. )
each entry gets a "special" field where one can quickly get what is so special about this location,
(like what kind of object could usualy be found at such a place) or what is so special about this npc, what is he good for ? is he a provider for some rare resource or objects ?
the special field could help denote quickly what's important.

one additional merit of having such a codex, is that it would allow to unclutter _a lot_ the other usual panels  ( 'examine' and such ). imho, as it stand currently +60% of infos in those could be removed. it should bring a lot of breathe space. for example when i find a backpack in game, i just whant to know how much stuff i can put in it, if it's sturdy or flimsy, in good shape or damaged, filthy or clean and similar info, a quick descrition and that's it. most of the other details (which can be important too) could be indicated in the object dedicated entry in the codex. (how to repair it, what component you get from disassembling it, what material it's made with, if it conduct electricity or not , if it's water tight, etc ,etc ,etc.. ).

### Additional context

that idea was made originaly for cdda by myself. i import it here in an attempt to revive it, since i refuse to work with cdda anymore. to stay on the pragmatic side of things, i'll just say that i disagree with the approach they are taking on many fronts, in design and user experience, making redundants panels, redundant infos everywhere, cluttered panels, scattered infos, and generaly messy and inconsistent UI ). i was the one that brought the new switchable panels system  (among other contributions, like retrodays+ tileset and also the animation trick of flipping sprites left or right depending the direction, and some more).  for the records, my original idea, and the working prototype i had was quite different, better looking. i don't like what they've done with their own sauce/ implementation of it, to say the least.

**Comment by chaosvolt** [comment]
Again belated response since I missed this until it just abruptly came back into my feed: The idea of adding lore snippets to buildings and gameplay mechanics seems interesting, especially if that can be JSONized so that mods can inject their own lore discoveries into locations, quests, or other things they mod in. Though in lieu of JSONization, it could probably be hacked together in LUA too since BN has that again.

That said, one issue with lore things is that there's a clear, deliberate divide between the actual background to the game and what the player can encounter. There's a lot of conflicting stories, missing info, and overt misinformation that can be picked up in various snippets. Representing the player-character's prior knowledge of what actually happened, and adapting it to what new info they find out, would be a giant pain in the ass to implement, and avoiding that problem would require limiting codex info to an extremely pragmatic, neutral "I don't care who started it, I only care about who's currently trying to eat my face" perspective on what info they take in.

That said, the divide between actual lore, in-game misinformation, and the fun bonus category of "actual lore but only as an excuse for game mechanics we" has caused some fuckiness in the past. Interactions with that last one have caused arguments over things like healing rate being unrealistic vs the blob-magic angle, plus the current in-lore concept of haywire robots is still influenced by the old "reprogramed AI to aggressively contain the growing hordes" lore, which was initially intimately tied to the "oops they fail to recognize zombies as targets" thing that existed purely as an excuse for why turrets only shot at the player, back in the old days before monsters had proper support for infighting.

I don't recall of any direct clashes between true game lore and in-game misinformation directly affecting contributions, but it remains a possibility. I have seen someone (jokingly?) float the idea on the discord of encountering the Emperor of China as an in-game NPC, based off a zero-day newspaper snippet implying said person exists (despite "having an emperor is kinda cringe" being the main thing both the PRC and ROC have in common)...
